source/val/validate_memory.cpp

```diff
@@ -1356,27 +1356,46 @@ spv_result_t ValidateAccessChain(ValidationState_t& _,
     auto index_type = _.FindDef(cur_word_instr->type_id());
     if (!index_type || spv::Op::OpTypeInt != index_type->opcode()) {
       return _.diag(SPV_ERROR_INVALID_ID, inst)
-             << "Indexes passed to " << instr_name
+             << "Indexes passed to Op" << spvOpcodeString(opcode)
              << " must be of type integer.";
     }
     switch (type_pointee->opcode()) {
       case spv::Op::OpTypeMatrix:
       case spv::Op::OpTypeVector:
       case spv::Op::OpTypeCooperativeMatrixNV:
       case spv::Op::OpTypeCooperativeMatrixKHR:
-      case spv::Op::OpTypeArray:
       case spv::Op::OpTypeRuntimeArray: {
         // In OpTypeMatrix, OpTypeVector, spv::Op::OpTypeCooperativeMatrixNV,
         // OpTypeArray, and OpTypeRuntimeArray, word 2 is the Element Type.
         type_pointee = _.FindDef(type_pointee->word(2));
         break;
       }
+      case spv::Op::OpTypeArray: {
+        // If using a descriptor array, make sure the outer array is not OOB
+        if (spv::Op::OpConstant == cur_word_instr->opcode()) {
+          const uint32_t cur_index = cur_word_instr->word(3);
+          auto array_length_id = _.FindDef(type_pointee->word(3));
+          if (spv::Op::OpConstant == array_length_id->opcode()) {
+            const uint32_t array_length = array_length_id->word(3);
```

Comment:
while the `OpAccessChain` indexes are required to be 32-bit int scalar but can be negative

I realize from the failing `spirv-opt` tests is the `OpTypeArray` length operand can be a 64-bit length